src/_cffi_src/build_commoncrypto.py

```diff
@@ -0,0 +1,32 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import, division, print_function
+
+from _cffi_src.utils import build_ffi_for_binding
+
+
+_module_prefix = "_cffi_src.commoncrypto."
+_modules = [
+    "cf",
+    "common_digest",
+    "common_hmac",
+    "common_key_derivation",
+    "common_cryptor",
+    "common_symmetric_key_wrap",
+    "secimport",
+    "secitem",
+    "seckey",
+    "seckeychain",
+    "sectransform",
+]
+
+ffi = build_ffi_for_binding(
+    so_name="_cryptography_commoncrypto",
```

Comment:
Yeah, the double cryptography seems wrong. I also feel like these should be
in hazmat/bindings/

On Sun, Jun 7, 2015 at 9:57 AM, Paul Kehrer <notifications@github.com>
wrote:

> In src/_cffi_src/build_commoncrypto.py
> <https://github.com/pyca/cryptography/pull/1986#discussion_r31877016>:
>
> > +_modules = [
> > +    "cf",
> > +    "common_digest",
> > +    "common_hmac",
> > +    "common_key_derivation",
> > +    "common_cryptor",
> > +    "common_symmetric_key_wrap",
> > +    "secimport",
> > +    "secitem",
> > +    "seckey",
> > +    "seckeychain",
> > +    "sectransform",
> > +]
> > +
> > +ffi = build_ffi_for_binding(
> > +    so_name="_cryptography_commoncrypto",
>
> They do. Right now to import this module you do from
> cryptography._cryptography_commoncrypto import ffi, lib. Arguably
> _cryptography becomes an unneeded piece. It was added partially to
> disambiguate in case the so doesn't get installed to the right place for
> reason.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1986/files#r31877016>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084